components/server/src/server.rs

```diff
@@ -1015,6 +1024,65 @@ impl<ER: RaftEngine> TiKVServer<ER> {
             });
     }
 
+    fn init_storage_stats_task(&self, engines: Engines<RocksEngine, ER>) {
+        let config_disk_capacity: u64 = self.config.raft_store.capacity.0;
+        let store_path = self.store_path.clone();
+        let snap_mgr = self.snap_mgr.clone().unwrap();
+        //TODO wal size ignore?
+        self.background_worker
+            .spawn_interval_task(DEFAULT_STORAGE_STATS_INTERVAL, move || {
+                if disk::get_disk_reserved() == 0 {
+                    return;
+                }
+                let disk_stats = match fs2::statvfs(&store_path) {
+                    Err(e) => {
+                        error!(
+                            "get disk stat for kv store failed";
+                            "kv path" => store_path.to_str(),
+                            "err" => ?e
+                        );
+                        return;
+                    }
+                    Ok(stats) => stats,
+                };
+                let disk_cap = disk_stats.total_space();
+                let snap_size = snap_mgr.get_total_snap_size().unwrap();
+
+                let kv_size = engines
+                    .kv
+                    .get_engine_used_size()
+                    .expect("get kv engine size");
+
+                let raft_size = engines
+                    .raft
+                    .get_engine_size()
+                    .expect("get raft engine size");
+
+                let used_size = snap_size + kv_size + raft_size;
+                let capacity = if config_disk_capacity == 0 || disk_cap < config_disk_capacity {
+                    disk_cap
+                } else {
+                    config_disk_capacity
+                };
+
+                let mut available = capacity.checked_sub(used_size).unwrap_or_default();
+                available = cmp::min(available, disk_stats.available_space());
+                if available <= disk::get_disk_reserved() {
```

Comment:
We have already reserved disk space by creating a placeholder file, adding this check along with the placeholder file will make the reserved size double /cc @hicqu , see: 
https://github.com/tikv/tikv/blob/a3da6cec14857de021da6ed14353ae461303047d/components/server/src/server.rs#L400-L420

Reply:
right, the place holder file we will deal with in another pr.

Reply:
Plz leave a TODO for now.

Reply:
ok. done